esmvaltool/cmorizers/data/cmorizer.py

```diff
@@ -0,0 +1,529 @@
+                           self.datasets, self.rawobs)
+        logger.info("Processing datasets %s", datasets)
+
+        # loop through tier/datasets to be cmorized
+        failed_datasets = []
+        for dataset in datasets:
+            if not self.format_dataset(dataset, start, end, install):
+                failed_datasets.append(dataset)
+
+        if failed_datasets:
+            raise Exception(
+                f'Format failed for datasets {" ".join(failed_datasets)}')
+
+    @staticmethod
+    def has_downloader(dataset):
+        """Check if a given datasets has an automatic downloader.
+
+        Parameters
+        ----------
+        dataset : str
+            Name of the dataset to check
+
+        Returns
+        -------
+        str
+            'Yes' if the downloader exist, 'No' otherwise
+        """
+        try:
+            importlib.import_module(
+                f'.{dataset.lower().replace("-", "_")}',
+                package='esmvaltool.cmorizers.data.downloaders.datasets')
+            return 'Yes'
+        except ImportError:
+            return 'No'
+
+    def _assemble_datasets(self):
+        """Get my datasets as dictionary keyed on Tier."""
+        # check for desired datasets only (if any)
+        # if not, walk all over rawobs dir
+        # assume a RAWOBS/TierX/DATASET input structure
+
+        # get all available tiers in source dir
+        tiers = ['Tier{}'.format(i) for i in [2, 3]]
+        tiers = [
+            tier for tier in tiers
+            if os.path.exists(os.path.join(self.rawobs, tier))
+        ]
+        datasets = []
+        if self.datasets:
+            return self.datasets
+        for tier in tiers:
+            for dataset in os.listdir(os.path.join(self.rawobs, tier)):
+                datasets.append(dataset)
+
+        return datasets
+
+    def format_dataset(self, dataset, start, end, install):
+        """Format a single dataset.
+
+        Parameters
+        ----------
+        dataset: str
+            Dataset name
+        start: datetime
+            Start of the period to format
+        end: datetime
+            End of the period to format
+        install: bool
+            If True, automatically copies the data to the final location if
+            there is no
+        """
+        reformat_script_root = os.path.join(
+            os.path.dirname(os.path.abspath(__file__)), 'formatters',
+            'datasets', self._dataset_to_module(dataset))
+        tier = self._get_dataset_tier(dataset)
+        if tier is None:
+            logger.error('Data for %s not found', dataset)
+            return False
+
+        # in-data dir; build out-dir tree
+        in_data_dir = os.path.join(self.rawobs, tier, dataset)
+        logger.info("Input data from: %s", in_data_dir)
+        out_data_dir = os.path.join(self.output_dir, tier, dataset)
+        logger.info("Output will be written to: %s", out_data_dir)
+        if not os.path.isdir(out_data_dir):
```

Comment:
Just a thought: the current output directory is in the form `esmvaltool_output/data_download_YYYYMMDD_HHMMSS/TierX/DATASET` and I find the "data_download" not very telling. In some cases, one uses only the `format` function for which no download is needed. Also, when use the new cmorizer functions multiple time (like I'm doing), it is not so esay to quickly find which directory corresponds to which run. This is already the case which the previous `cmorize_obs` function. Could we instead rename the output directories to something more user-friendly like: `esmvaltool_output/cmorize_DATASET_YYYYMMDD_HHMMSS/TierX/DATASET`?  Or prepare instead of cmorize? This would be the equivalent to `esmvaltool_output/recipe_python_YYYYMMDD_HHMMSS` where the name of the recipe is given in the path.

Reply:
If fixed it so it is `data_download`, `data_formatting` or `data_preparation` according to the command isued. I am not sure about adding the dataset to the folder, as we can potentially have more than one in a call